setting-up-ocarina.md

```diff
@@ -0,0 +1,120 @@
+
+```
+{"MAJORA_DOMAIN": "https://majora.covid19.climb.ac.uk/", "MAJORA_USER": "your-username", "MAJORA_TOKEN": "OAUTH", "CLIENT_ID": "your-client-id", "CLIENT_SECRET": "your-client-secret"}
+```
+
+Where:
+* `MAJORA_DOMAIN` points to either the domain of the real, or test Majora. Do not miss the `/` at the end of the URL.
+* `MAJORA_USER` is your username on the appropriate Majora instance
+* `MAJORA_TOKEN` must be set to `OAUTH`
+* `CLIENT_ID` is the ocarina Client ID registered with majora [accessible at this majora page](https://majora.covid19.climb.ac.uk/o/applications/)
+* `CLIENT_SECRET` is the ocarina Secret ID registered with majora [accessible at this majora page](https://majora.covid19.climb.ac.uk/o/applications/)
+
+OAuth has been the standard way to authenticate since September 2020 but legacy API keys are still supported, [instructions here.](getting-api-keys)
+
+### 3b. Set your Ocarina credentials with a legacy API key
+
+[Generate a legacy API key](getting-api-keys) if you haven't already.
+
+Create a JSON file in your home directory named `.ocarina`. Note the starting dot.
+
+```
+{"MAJORA_DOMAIN": "https://majora.covid19.climb.ac.uk/", "MAJORA_USER": "your-username", "MAJORA_TOKEN": "YOUR_API_KEY", "CLIENT_ID": "", "CLIENT_SECRET": ""}
+```
+
+Where:
+* `MAJORA_DOMAIN` points to either the domain of the real, or test Majora. Do not miss the `/` at the end of the URL.
+* `MAJORA_USER` is your username on the appropriate Majora instance
+* `MAJORA_TOKEN` taken from [this majora page](https://majora.covid19.climb.ac.uk/keys/list/)
+
+OAuth has been the standard way to authenticate since September 2020 but legacy API keys are still supported, [instructions here.](getting-api-keys)
+
+### 4. Submit an ocarina query and generate an OAuth token
+
+Activate your Ocarina conda environment
+
+```
+conda activate ocarina
+```
+
+Attempt to access a biosample with the following command
+
+```
+ocarina --oauth get biosample --central-sample-id *Insert COG ID here*
+```
+
+You should see a message asking you to 'request a grant'. Open the link (or copy it into your a browser on your local machine) to see an approval screen like this:
+
+![image](images/oauth_example.png)
+
+Then click "Authorize" to confirm that you have indeed requested an OAuth token and you should see a page which looks like this apart from the text boxes will be populated:
+
+![image](images/ocarina_example.png)
+
+Now click the copy button next to the the upper text box to copy the callback URL.
+
+Now paste the copied return URL into Ocarina which should now return your requested dataview.
+
+### 5. Refreshing OAuth tokens
+
+Now you have fully set up OAuth and Ocarina in order to keep using the same token you can refresh it with the command `ocarina oauth refresh`, these tokens expire after **10 hours after which you will have to repeat the process of generating a token by following step 4 again.**
```

Comment:
make it a bit clearer that refreshing the token should be done before the 10 others "otherwise you will have to repeat the process from Step 4"